go.mod

```diff
@@ -4,6 +4,32 @@ go 1.24.0
 
 toolchain go1.24.5
 
+// These are needed for the OTE tests.
+replace (
+	github.com/onsi/ginkgo/v2 => github.com/openshift/onsi-ginkgo/v2 v2.6.1-0.20241205171354-8006f302fd12
+	k8s.io/apiserver => k8s.io/apiserver v0.33.3
+	k8s.io/kubelet => k8s.io/kubelet v0.33.3
+	k8s.io/kubernetes => k8s.io/kubernetes v1.33.3
+)
+
+// Pin all k8s.io packages to consistent versions to avoid dependency conflicts
+replace (
+	github.com/google/gnostic-models => github.com/google/gnostic-models v0.6.9
+	go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc => go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc v0.59.0
+	k8s.io/api => k8s.io/api v0.33.3
+	k8s.io/apimachinery => k8s.io/apimachinery v0.33.3
+	k8s.io/client-go => k8s.io/client-go v0.33.3
+	k8s.io/code-generator => k8s.io/code-generator v0.33.3
+	k8s.io/component-base => k8s.io/component-base v0.33.3
+	k8s.io/cri-api => k8s.io/cri-api v0.33.3
+	k8s.io/cri-client => k8s.io/cri-client v0.33.3
+	k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20250318190949-c8a335a9a2ff
+	k8s.io/kubectl => k8s.io/kubectl v0.33.3
+	k8s.io/utils => k8s.io/utils v0.0.0-20250604170112-4c0f3b243397
+	sigs.k8s.io/controller-runtime => sigs.k8s.io/controller-runtime v0.20.1
+	sigs.k8s.io/structured-merge-diff/v4 => sigs.k8s.io/structured-merge-diff/v4 v4.6.0
+)
+
```

Comment:
I have concerns with replacing the k8s modules. These are particularly controlled to certain versions. OpenShift v4.20 should be using k8s 1.33, etc. It is not clear to me why we would need to override/replace them like this.